File: PWRSyllabus.Core/Entities/FieldOfStudy.cs
using System.Collections.Generic;

namespace PWRSyllabus.Core.Entities
{
    public class FieldOfStudy : BaseEntity
    {
        public string Specialization { get; set; }
        public string Name { get; set; }
        public List<FieldOfStudyDiscipline> FieldOfStudyDisciplines { get; set; }
        public Faculty Faculty { get; set; }
        public int FacultyId { get; set; }
        public List<StudyProgram> StudyPrograms { get; set; }
    }
}

File: PWRSyllabus.API/ClientApp/src/models/FieldOfStudy.ts
export default interface FieldOfStudy {
    id: number;
    name: string;
    specialization: string;
    faculty: string;
}

const DefaultFieldOfStudy: FieldOfStudy = {
    id: 0,
    name: '',
    specialization: '',
    faculty: '',
};

export {DefaultFieldOfStudy, FieldOfStudy};
